ballerina/model-provider.bal

```diff
@@ -103,3 +105,117 @@ public type ModelProvider distinct isolated client object {
     isolated remote function chat(ChatMessage[] messages, ChatCompletionFunctions[] tools = [], string? stop = ())
         returns ChatAssistantMessage|LlmError;
 };
+
+# Represents configuratations of WSO2 provider.
+#
+# + serviceUrl - The URL for the WSO2 AI service
+# + accessToken - Access token for accessing WSO2 AI service
+public type Wso2ProviderConfig record {|
+    string serviceUrl;
+    string accessToken;
+|};
+
+# Configurable for WSO2 provider.
+configurable Wso2ProviderConfig? wso2ProviderConfig = ();
+
+# WSO2 model provider implementation that provides chat completion capabilities using WSO2's AI services.
+public isolated distinct client class Wso2ModelProvider {
+    *ModelProvider;
+    private final intelligence:Client llmClient;
+
+    # Initializes a new `WSO2ModelProvider` instance.
```

Comment:
Shall we add the doc saying, this is based on `AzureOpenAI`. Imo we need to give that information to user.
wdyt?

Reply:
ok. what I felt was that users may be interested to know the ModelProvider somewhere in the docs, because results are vary depending on the model providers